qdev_wrappers/tests/dataset/test_doNd.py

```diff
@@ -0,0 +1,165 @@
+"""
+These are the basic black box tests for the doNd functions.
+"""
+
+from qdev_wrappers.dataset.doNd import do0d, do1d, do2d
+from typing import Tuple, List, Optional
+from qcodes.instrument.parameter import Parameter
+from qcodes import config, new_experiment
+from qcodes.utils import validators
+
+import pytest
+
+config.user.mainfolder = "output"  # set ouput folder for doNd's
+new_experiment("doNd-tests", sample_name="no sample")
+
+
+@pytest.fixture()
+def _parameters():
+
+    _param = Parameter('simple_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1)
+
+    _paramComplex = Parameter('simple_complex_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1 + 1j,
+                   vals=validators.ComplexNumbers())
+
+    _param_set = Parameter('simple_setter_parameter',
+                       set_cmd=None,
+                       get_cmd=None)
+
+    return _param, _paramComplex, _param_set
+
+
+def _param_func(_p):
+    """
+    A private utility function.
+    """
+    _new_param = Parameter('modified_parameter',
+                           set_cmd= None,
+                           get_cmd= lambda: _p.get()*2)
+    return _new_param
+
+
+@pytest.fixture()
+def _param_callable(_parameters):
```

Comment:
I am confused by this comment. I do implement this test with an arbitrary callable, but just to be fair to the context I created a function which manipulates a QCoDeS parameter; not because I think Callables are related to QCoDeS parameters :)  

The main purpose here to test if doNd work with an arbitrary function as its positional argument. That is it. 

I am also missing the point of your example; do we really need to create a callable object?

Reply:
Ok, great, I simply misunderstood you there, because going through a qcodes parameter does not seem like the most obvious solution to me. I was wondering why you did not simply do this:
```python
@pytest.fixture
def callable():
    return lambda: 1
```
This does exactly the same as currently.

Concerning my sample code in my previous comment:
I created a callable class so that we can attach a state that we can readout, here the time stamps when it has been called. This could then help us to write a test which verifies the order in which callables and parameters have been called.

Reply:
I have decided the let this function and the corresponding test stays as it is; but they can later be improved on QCoDeS site, if needed.